### PR TITLE
Utils restructure

### PR DESCRIPTION
**Pull request by remcohaszing**
I think these patches uncover some issues with the utils. Also the utils are untested.

I would like to make tests for them and fix those issues if I'm sure I've got the documentation right.

### CODE REVIEW

File: rules/utils/utils.js [hunk 1]
```diff
@@ -12,190 +69,430 @@ function getCallingIdentifier(calleeObject) {
...
+ *     ^^^^^^^
+ * app.constant('KITTENS', function() {})
+ *     ^^^^^^^^
+ * app.value('KITTENS', function() {})
+ *     ^^^^^
+ * ```
+ *
+ * The following are not considered services
+ * ```js
+ * $provide.factory('kittenService', function() {})
```
Comment (Gillespie59) [inline rules/utils/utils.js]: If these are not considered as service, I think this is a mistake. We should update this isAngularServiceDeclaration method in order to support these exta syntaxes.

Comment (remcohaszing) [inline rules/utils/utils.js]: That's why I wanted someone else to take a look at the docs I added and see if they are correct. If these docs are correct I would like to add tests and fix the issues I marked with `@todo FIXME`.

**Comment by Gillespie59** [comment]
for me, the doc looks good.

**Comment by tilmanpotthof** [comment]
looks good for me too. I'll merge it.

**Pull request merged**